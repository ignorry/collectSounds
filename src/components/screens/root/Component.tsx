import React from "react";
import styled from "styled-components"
import { Route, Routes } from "react-router-dom";

import Navigation from "./components/Navigation";
import Error from "./components/Error";

import Search from "../search";
import Video from "../video";
import Playlist from "../playlist";
import Channel from "../channel";
import Saved from "../saved";
import Queue from "../queue";
import Settings from "../settings";

const BASE_URL = !process.env.NODE_ENV || process.env.NODE_ENV === 'development' ? "" : "/collectSounds/";

export const Content = styled.div`
  overflow: hidden;
  display: flex;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;

  @media ${ ({ theme }) => theme.media.small } {
    display: block;
  }
`;

export const Nav = styled.div`
  padding: ${ ({ theme }) => `${ theme.gaps.smallest }rem` } 0;
  background: ${ ({ theme }) => theme.colors.bgSecondary };
  z-index: ${ ({ theme }) => theme.order.footer };
  border-right: 1px solid ${ ({ theme }) => theme.colors.bg };

  @media ${ ({ theme }) => theme.media.small } {
    position: absolute;
    right: 0;
    bottom: 0;
    left: 0;
    border-right: none;
  }
`;

const Component: React.FC = () => (
  <Content>
    <Nav>
      <Navigation/>
    </Nav>
    <Routes>
      <Route path={`${ BASE_URL || '/' }`} element={ <Search/> }/>
      <Route path={`${ BASE_URL }search/video/:id`} element={ <Video/> }/>
      <Route path={`${ BASE_URL }search/playlist/:id`} element={ <Playlist/> }/>
      <Route path={`${ BASE_URL }search/channel/:id`} element={ <Channel/> }/>
      <Route path={`${ BASE_URL }search`} element={ <Search/> }/>
      <Route path={`${ BASE_URL }saved/video/:id`} element={ <Video/> }/>
      <Route path={`${ BASE_URL }saved/playlist/:id`} element={ <Playlist/> }/>
      <Route path={`${ BASE_URL }saved/channel/:id`} element={ <Channel/> }/>
      <Route path={`${ BASE_URL }saved`} element={ <Saved/> }/>
      <Route path={`${ BASE_URL }queue`} element={ <Queue/> }/>
      <Route path={`${ BASE_URL }settings`} element={ <Settings/> }/>
    </Routes>
    <Error />
  </Content>
);

export default Component;