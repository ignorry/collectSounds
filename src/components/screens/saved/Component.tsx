import React, { useEffect, useState } from "react";
import styled from "styled-components"
import { SearchQueries, searchSaved } from "../../../lib/redux/searchSaved";
import { Content } from "../../../models/content";
import { useNavigate } from "react-router-dom";
import { useIntl } from "react-intl";
import queryString from "query-string";

import Filters from "./components/Filters";
import List, { Props as ListProps } from "../../content/List";
import Spinner from "../../primitives/Spinner";
import Tabs from "../../primitives/Tabs";
import { useDispatch } from "react-redux";

const SearchContainer = styled.div`
  background: ${ ({ theme }) => theme.colors.bgSecondary };
  padding-bottom: ${ ({ theme }) => `${ theme.font.primary.size*1.1 + theme.gaps.small*2 }rem`};
  margin-bottom: ${ ({ theme }) => `-${ theme.font.primary.size*1.1 + theme.gaps.small*2 }rem`};
`;

const TabsContainer = styled.div`
  max-width: ${ ({ theme }) => `${ theme.contentWidth }rem`};
  width: 100%;
  margin: auto;
`;

const SpinnerContainer = styled.div`
  padding: ${ ({ theme }) => `${ theme.gaps.big }rem` };
`;

const Component: React.FC = () => {
  const dispatch = useDispatch();
  const navigate = useNavigate();
  const intl = useIntl();
  const queryStr = queryString.parse( location.search );

  const [isVideos, setIsVideos] = useState<boolean>( queryStr.isVideos === 'true' );
  const [queries, setQueries] = useState<SearchQueries | null>( 
    Object.fromEntries( Object.entries( queryStr ).filter( item => item[0] !== 'isVideos' && item[0] !== 'index' )));
  const [res, setRes] = useState<Array<Content> | null>( null );

  const handleSearch = ( queries: SearchQueries ) => {
    setQueries( queries );

    if ( /^https:\/\/www.youtube.com\/watch\?v=/.test( queries.q ) )
      navigate( `/saved/video/${ queries.q.split( '=' )[1].split( '&' )[0] }` );
    if ( /^https:\/\/www.youtube.com\/playlist\?list=/.test( queries.q ) )
      navigate( `/saved/playlist/${ queries.q.split( '=' )[1].split( '&' )[0] }` );

    const stringified = queryString.stringify( { ...queries, isVideos } );
    window.history.pushState('', '', `${ window.location.origin }${ window.location.pathname }?${ stringified }` );

    setRes( null );
  }

  const tab = ( index: number ) => {
    setIsVideos( index ? true : false );
    const stringified = queryString.stringify( { ...queries, isVideos: index ? true : false } );
    window.history.pushState('', '', `${ window.location.origin }${ window.location.pathname }?${ stringified }` );
  };

  useEffect( () => {
    const res: any = isVideos ? dispatch( searchSaved( { ...queries, type: 'video' } ) as any )
      : dispatch( searchSaved( queries ) as any );

    setRes( res );
  }, [queries, isVideos]);

  const list = res ? <List items={ res.map( item => ({
    content: item,
    callback: () => navigate( `/saved/${item.type}/${item.id}` )
  })) } emptyMessage={ intl.formatMessage({ id: 'emptyRes' })}/> : <SpinnerContainer><Spinner/></SpinnerContainer>;

  const tabs = [
    {
      label: intl.formatMessage({ id: 'all' }),
      content: list
    },
    {
      label: intl.formatMessage({ id: 'videos' }),
      content: list
    }
  ];

  return (
    <div>
      <SearchContainer>
        <Filters queries={ queries } callback={ handleSearch } isVideo={ isVideos }/>
      </SearchContainer>
      <TabsContainer>
        <Tabs tabs={ tabs } onChange={ tab } active={ isVideos ? 1 : 0 }/>
      </TabsContainer>
    </div>
  );
}

export default Component;