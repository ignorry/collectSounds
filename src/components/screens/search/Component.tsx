import React, { useEffect, useState } from "react";
import styled from "styled-components"
import { getPopularVideos, getSearch, SearchFilters } from "../../../lib/api";
import { Content } from "../../../models/content";
import { useNavigate } from "react-router-dom";
import { useIntl } from "react-intl";
import queryString from "query-string";

import SearchQueries from "./components/SearchQueries";
import List, { Props as ListProps } from "../../content/List";
import Spinner from "../../primitives/Spinner";

const BASE_URL = !process.env.NODE_ENV || process.env.NODE_ENV === 'development' ? "" : "/collectSounds/";

const SearchContainer = styled.div`
  background: ${ ({ theme }) => theme.colors.bgSecondary };
`;

const ListContainer = styled.div`
  margin: auto;
  max-width: ${ ({ theme }) => `${ theme.contentWidth }rem` };
  padding: ${ ({ theme }) => `${ theme.gaps.big }rem` };
`;

const SpinnerContainer = styled.div`
  padding: ${ ({ theme }) => `${ theme.gaps.big }rem` };
`;

const Component: React.FC = () => {
  const navigate = useNavigate();
  const intl = useIntl();

  const [queries, setQueries] = useState<SearchFilters | null>( queryString.parse( location.search ) );
  const [res, setRes] = useState<Array<Content> | null>( null );

  const handleSearch = ( queries: SearchFilters ) => {
    setQueries( queries );

    if ( /^https:\/\/www.youtube.com\/watch\?v=/.test( queries.q ) )
      navigate( `${ BASE_URL || '/' }search/video/${ queries.q.split( '=' )[1].split( '&' )[0] }` );
    if ( /^https:\/\/www.youtube.com\/playlist\?list=/.test( queries.q ) )
      navigate( `${ BASE_URL || '/' }search/playlist/${ queries.q.split( '=' )[1].split( '&' )[0] }` );

    const stringified = queryString.stringify( queries );
    window.history.pushState('', '', `${ window.location.origin }${ window.location.pathname }?${ stringified }` );

    setRes( null );
  }

  useEffect( () => {
    if ( queries )
      getSearch( queries ).then( items => setRes( items.filter( item => item ) ) );
  }, [queries]);

  useEffect( () => {
    if ( queries )
      getSearch( queries ).then( items => setRes( items.filter( item => item ) ) );
    else
      getPopularVideos().then( items => setRes( items.filter( item => item) ) );
  }, []);

  const list = res ? <List items={ res.map( item => ({
    content: item,
    callback: () => navigate( `${ BASE_URL || '/' }search/${item.type}/${item.id}` )
  })) } emptyMessage={ intl.formatMessage({ id: 'emptyRes' })}/> : <SpinnerContainer><Spinner/></SpinnerContainer>;

  return (
    <div>
      <SearchContainer>
        <SearchQueries queries={ queries } callback={handleSearch}/>
      </SearchContainer>
      <ListContainer>
        { list }
      </ListContainer>
    </div>
  );
}

export default Component;