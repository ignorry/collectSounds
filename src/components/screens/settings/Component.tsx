import React from "react";
import { connect, ConnectedProps, useDispatch } from "react-redux";
import styled from "styled-components"
import { useIntl } from "react-intl";
import { setDarkTheme } from "../../../redux/slices/options";

import Button from "../../primitives/Button";
import Label from "../../primitives/Label";
import ShowQR from "./components/ShowQR";
import ScanQR from "./components/ScanQR";
import Conncted from "./components/Connected";

const Container = styled.div`
  margin: auto;
  max-width: ${ ({ theme }) => `${ theme.contentWidth }rem`};
  width: 100%;
  padding: ${ ({ theme }) => `${ theme.gaps.big }rem` };
  display: flex;
  flex-direction: column;
  gap: ${ ({ theme }) => `${ theme.gaps.big }rem` };
`;

const mapState = ( state: any ) => {
  return {
    options: state.options,
  }
}

const connector = connect( mapState );

type Props = ConnectedProps<typeof connector>;

const Component: React.FC<Props> = ( props: Props ) => {
  const intl = useIntl()
  const dispatch = useDispatch();

  return (
    <Container>
      <Button callback={() => dispatch( setDarkTheme( ! props.options.darkTheme ))} text={ intl.formatMessage({ id: 'theme' }) } pic="theme" />

      <ScanQR/>
      <ShowQR peer={ props.options.peer }/>
      <Conncted peers={ props.options.connectedPeers }/>
    </Container>
  );
}

export default connector( Component );