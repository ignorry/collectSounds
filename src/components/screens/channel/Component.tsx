import React, { useEffect, useState } from "react";
import { connect, ConnectedProps, useDispatch } from "react-redux";
import styled from "styled-components"
import { Channel, Playlist } from "../../../models/content";
import { useIntl } from "react-intl";
import { addItem, addPlaylistToParent, addVideoToParent, deleteItem, getChannel, getPlaylist, updatePlaylist } from "../../../redux/slices/saved";
import { getChannelById, getChannelPlaylists, getChannelVideos, getPlaylistById, getPlaylistVideos } from "../../../lib/api";
import { setMessage } from "../../../redux/slices/errorMessage";

import Header from "../../primitives/Header";
import Content from "./components/Content";
import Spinner from "../../primitives/Spinner";
import { fetchPlaylistItems } from "../../../lib/api/fetch";

const SpinnerContainer = styled.div`
  padding: ${ ({ theme }) => `${ theme.gaps.big }rem` };
`;

const mapState = ( state: any ) => {
  return {
    saved: state.saved,
  }
}

const connector = connect( mapState );

type ReduxProps = ConnectedProps<typeof connector>;

type Props = ReduxProps & {
  id: string | null;
}

const Component: React.FC<Props> = ( props: Props ) => {
  const dispatch = useDispatch();
  const intl = useIntl();

  const [channel, setChannel] = useState<Channel | null>( null );
  const [saved, setSaved] = useState<boolean>( false );

  const toggleSaved = () => {
    const current = ! saved;

    if ( channel )
      if ( current )
        dispatch( addItem( [ channel ] ) );
      else
        dispatch( deleteItem( [ channel.id ] ) );
    
    setSaved( ! saved );
  }

  useEffect( () => {
    if ( ! props.id ) {
      dispatch( setMessage( intl.formatMessage({ id: 'emptyMessage' }) ) );
      return;
    }

    const savedChannel: Channel | undefined = getChannel( props.saved, props.id );

    if ( savedChannel ) {
      setSaved( true );
      setChannel( savedChannel );
    } else 
      getChannelById( props.id ).then( items => items[0] ? setChannel( items[0] ) : null );
  }, []);

  useEffect( () => {
    if ( channel && ! channel.videos ) {
      getChannelVideos( channel.id ).then( items => {
        if ( saved ) {
          if ( items ) { items.forEach( video => {
            dispatch( addVideoToParent( {
              id: channel.id,
              video,
            }));
          })}
        }
        setChannel( { ...channel, videos: items.map( video => ({ id: video.id, item: video }))});
      });
    }

    if ( channel && ! channel.playlists ) {
      getChannelPlaylists( channel.id ).then( items => {
        if ( saved ) {
          if ( items ) { items.forEach( playlist => {
            dispatch( addPlaylistToParent( {
              id: channel.id,
              playlist,
            }));
          })}
        } else {
          setChannel( { ...channel, playlists: items.map( video => ({ id: video.id, item: video }))});
        }
      });
    }
  }, [channel]);

  const content = channel ? <Content
    channel={ channel }
    toggleSaved={ toggleSaved }
    saved={ saved }
  /> : <SpinnerContainer><Spinner/></SpinnerContainer>;

  return (
    <>
      <Header/>
      { content }
    </>
  );
}

export default connector( Component );