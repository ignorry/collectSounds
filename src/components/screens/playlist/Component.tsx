import React, { useEffect, useState } from "react";
import { connect, ConnectedProps, useDispatch } from "react-redux";
import styled from "styled-components"
import { Playlist } from "../../../models/content";
import { useIntl } from "react-intl";
import { addItem, addVideoToParent, deleteItem, getPlaylist, updatePlaylist } from "../../../redux/slices/saved";
import { getPlaylistById, getPlaylistVideos } from "../../../lib/api";
import { setMessage } from "../../../redux/slices/errorMessage";

import Header from "../../primitives/Header";
import Content from "./components/Content";
import Spinner from "../../primitives/Spinner";
import { fetchPlaylistItems } from "../../../lib/api/fetch";

const SpinnerContainer = styled.div`
  padding: ${ ({ theme }) => `${ theme.gaps.big }rem` };
`;

const mapState = ( state: any ) => {
  return {
    saved: state.saved,
  }
}

const connector = connect( mapState );

type ReduxProps = ConnectedProps<typeof connector>;

type Props = ReduxProps & {
  id: string | null;
}

const Component: React.FC<Props> = ( props: Props ) => {
  const dispatch = useDispatch();
  const intl = useIntl();

  const [playlist, setPlaylist] = useState<Playlist | null>( null );
  const [saved, setSaved] = useState<boolean>( false );

  const toggleSaved = () => {
    const current = ! saved;

    if ( playlist )
      if ( current )
        dispatch( addItem( [ playlist ] ) );
      else
        dispatch( deleteItem( [ playlist.id ] ) );
    
    setSaved( ! saved );
  }

  useEffect( () => {
    if ( ! props.id ) {
      dispatch( setMessage( intl.formatMessage({ id: 'emptyMessage' }) ) );
      return;
    }

    const savedPlaylist: Playlist | undefined = getPlaylist( props.saved, props.id );

    if ( savedPlaylist ) {
      setSaved( true );
      setPlaylist( savedPlaylist );
    } else 
      getPlaylistById( props.id ).then( items => items[0] ? setPlaylist( items[0] ) : null );
  }, []);

  useEffect( () => {
    if ( playlist && ! playlist.videos ) {
      getPlaylistVideos( playlist.id ).then( items => {
        if ( saved ) {
          if ( items ) { items.forEach( video => {
            dispatch( addVideoToParent( {
              id: playlist.id,
              video,
            }));
          })}
        }
        setPlaylist( { ...playlist, videos: items.map( video => ({ id: video.id, item: video }))});
      });
    }
  }, [playlist]);

  const content = playlist ? <Content
    playlist={ playlist }
    toggleSaved={ toggleSaved }
    saved={ saved }
  /> : <SpinnerContainer><Spinner/></SpinnerContainer>;

  return (
    <>
      <Header/>
      { content }
    </>
  );
}

export default connector( Component );