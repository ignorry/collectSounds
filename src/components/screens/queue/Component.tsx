import React, { useEffect, useState } from "react";
import { connect, ConnectedProps, useDispatch } from "react-redux";
import styled from "styled-components"
import { SearchQueries, searchSaved } from "../../../lib/redux/searchSaved";
import { Content } from "../../../models/content";
import { useNavigate } from "react-router-dom";
import { useIntl } from "react-intl";
import queryString from "query-string";

import Header from "./components/Header";
import Spinner from "../../primitives/Spinner";
import Tabs from "../../primitives/Tabs";
import QueueList from "../../content/QueueList";
import { getVideo } from "../../../redux/slices/saved";

const List = styled.div`
  max-width: ${ ({ theme }) => `${ theme.contentWidth }rem`};
  width: 100%;
  margin: auto;
`;

const mapState = ( state: any ) => {
  return {
    saved: state.saved,
    queue: state.queue.data,
  }
}

const connector = connect( mapState );

type Props = ConnectedProps<typeof connector>;

const Component: React.FC<Props> = ( props: Props ) => {
  const intl = useIntl()
  const dispatch = useDispatch();

  const items = props.queue.map( ( item: any ) => ({ content: dispatch( getVideo( props.saved, item ) ), callback: () => {} }) );

  return (
    <div>
      <Header />
      <List>
        <QueueList items={ items } emptyMessage={ intl.formatMessage({ id: 'emptyList' }) }/>
      </List>
    </div>
  );
}

export default connector( Component );