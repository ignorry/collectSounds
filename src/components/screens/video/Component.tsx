import React, { useEffect, useState } from "react";
import { connect, ConnectedProps, useDispatch } from "react-redux";
import styled from "styled-components"
import { Video } from "../../../models/content";
import { useIntl } from "react-intl";
import { addItem, deleteItem, getVideo } from "../../../redux/slices/saved";
import { getVideoById } from "../../../lib/api";
import { setMessage } from "../../../redux/slices/errorMessage";

import Header from "../../primitives/Header";
import Content from "./components/Content";
import Spinner from "../../primitives/Spinner";

const SpinnerContainer = styled.div`
  padding: ${ ({ theme }) => `${ theme.gaps.big }rem` };
`;

const mapState = ( state: any ) => {
  return {
    saved: state.saved,
  }
}

const connector = connect( mapState );

type ReduxProps = ConnectedProps<typeof connector>;

type Props = ReduxProps & {
  id: string | null;
}

const Component: React.FC<Props> = ( props: Props ) => {
  const dispatch = useDispatch();
  const intl = useIntl();

  const [video, setVideo] = useState<Video | null>( null );
  const [saved, setSaved] = useState<boolean>( false );

  const toggleSaved = () => {
    const current = ! saved;

    if ( video )
      if ( current )
        dispatch( addItem( [ video ] ) );
      else
        dispatch( deleteItem( [ video.id ] ) );
    
    setSaved( ! saved );
  }

  useEffect( () => {
    if ( ! props.id ) {
      dispatch( setMessage( intl.formatMessage({ id: 'emptyMessage' }) ) );
      return;
    }

    const savedVideo: Video | undefined = getVideo( props.saved, props.id );

    if ( savedVideo ) {
      setSaved( true );
      setVideo( savedVideo );
    } else 
      getVideoById( props.id ).then( items => items[0] ? setVideo( items[0] ) : null );
  }, [props.saved]);

  const content = video ? <Content
    video={ video }
    toggleSaved={ toggleSaved }
    saved={ saved }
  /> : <SpinnerContainer><Spinner/></SpinnerContainer>;

  return (
    <>
      <Header/>
      { content }
    </>
  );
}

export default connector( Component );