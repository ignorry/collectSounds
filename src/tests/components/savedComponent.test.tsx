import React from "react";
import { render, screen } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { getTheme } from "../../lib/theme";
import GlobalStyles from "../../lib/styled/global";
import { Provider } from "react-redux";
import store from "../../redux/index";
import { LangProvider } from "../../lib/localization";
import { BrowserRouter } from "react-router-dom";
import { SearchResponse } from "../../models/api/search";
import { VideoIdResponse } from "../../models/api/video";

import Component from "../../components/screens/saved/Component";
import { addItem } from "../../redux/slices/saved";
import { Video } from "../../models/content";

const AllTheProviders = ( { children }: any ) => (
  <Provider store={store}>
    <ThemeProvider theme={ getTheme( true ) } >
      <BrowserRouter>
        <LangProvider file="saved">
          { children }
        </LangProvider>
      </BrowserRouter>
      <GlobalStyles/>
    </ThemeProvider>
  </Provider>
);

describe( 'saved Component', () => {
  it( 'renders video component', async () => {
    store.dispatch( addItem([{
      type: 'video',
    } as Video ]));

    const result = render(
      <Component />,
      { wrapper: AllTheProviders }
    );

    await new Promise( resolve => setTimeout( resolve, 100 ) );

    const label = screen.getByText( 'Videos' );

    expect( label ).toBeTruthy();
  });

  it( 'matches to snapshot', async () => {
    const result = render(
      <Component />,
      { wrapper: AllTheProviders }
    );

    await new Promise( resolve => setTimeout( resolve, 10 ) );

    expect( result ).toMatchSnapshot();
  });
});