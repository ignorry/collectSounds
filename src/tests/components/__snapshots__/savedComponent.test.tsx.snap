// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`saved Component matches to snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="sc-efBctP edPByx"
        >
          <div
            class="sc-kDDrLX uIMVe"
          >
            <div
              class="sc-iqcoie cVepWP"
            >
              <input
                class="sc-bczRLJ dldCsv"
                placeholder="search or paste a link"
                value=""
              />
            </div>
            <div
              class="sc-eCYdqJ FxPOW"
            >
              <button
                class="sc-hKMtZM fntRFW"
              >
                <svg
                  class="sc-gsnTZi gSmFuM"
                >
                  <use
                    href="http://localhost/collectSounds/icons.svg#filters"
                  />
                </svg>
                <p
                  class="sc-dkzDqf gsGEgs"
                >
                  Filters
                </p>
              </button>
              <div
                class="sc-jSMfEi jmXqhj"
              >
                <div
                  class="sc-iBkjds envIss"
                >
                  <button
                    class="sc-gKXOVf feeGsc"
                  >
                    <p
                      class="sc-dkzDqf jHUImv"
                    >
                      Addition
                    </p>
                  </button>
                  <ul
                    class="sc-ftvSup hiVoSg"
                  >
                    <li>
                      <button
                        class="sc-gKXOVf feeGsc"
                      >
                        <p
                          class="sc-dkzDqf jHUImv"
                        >
                          Date
                        </p>
                      </button>
                    </li>
                    <li>
                      <button
                        class="sc-gKXOVf feeGsc"
                      >
                        <p
                          class="sc-dkzDqf jHUImv"
                        >
                          Title
                        </p>
                      </button>
                    </li>
                  </ul>
                </div>
                <div
                  class="sc-iBkjds envIss"
                >
                  <button
                    class="sc-gKXOVf feeGsc"
                  >
                    <p
                      class="sc-dkzDqf jHUImv"
                    >
                      All
                    </p>
                  </button>
                  <ul
                    class="sc-ftvSup hiVoSg"
                  >
                    <li>
                      <button
                        class="sc-gKXOVf feeGsc"
                      >
                        <p
                          class="sc-dkzDqf jHUImv"
                        >
                          Video
                        </p>
                      </button>
                    </li>
                    <li>
                      <button
                        class="sc-gKXOVf feeGsc"
                      >
                        <p
                          class="sc-dkzDqf jHUImv"
                        >
                          Playlist
                        </p>
                      </button>
                    </li>
                    <li>
                      <button
                        class="sc-gKXOVf feeGsc"
                      >
                        <p
                          class="sc-dkzDqf jHUImv"
                        >
                          Channel
                        </p>
                      </button>
                    </li>
                  </ul>
                </div>
                <div
                  class="sc-iBkjds envIss"
                >
                  <button
                    class="sc-gKXOVf feeGsc"
                  >
                    <p
                      class="sc-dkzDqf jHUImv"
                    >
                      Any
                    </p>
                  </button>
                  <ul
                    class="sc-ftvSup hiVoSg"
                  >
                    <li>
                      <button
                        class="sc-gKXOVf feeGsc"
                      >
                        <p
                          class="sc-dkzDqf jHUImv"
                        >
                          Long
                        </p>
                      </button>
                    </li>
                    <li>
                      <button
                        class="sc-gKXOVf feeGsc"
                      >
                        <p
                          class="sc-dkzDqf jHUImv"
                        >
                          Medium
                        </p>
                      </button>
                    </li>
                    <li>
                      <button
                        class="sc-gKXOVf feeGsc"
                      >
                        <p
                          class="sc-dkzDqf jHUImv"
                        >
                          Short
                        </p>
                      </button>
                    </li>
                  </ul>
                </div>
                <div
                  class="sc-jqUVSM hgFwMx"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-cTQhss fRlOMF"
        >
          <div>
            <div
              class="sc-iTONeN kTEwDW"
            >
              <button
                class="sc-hKMtZM ixzCuJ"
              >
                <p
                  class="sc-dkzDqf gsGEgs"
                >
                  All
                </p>
              </button>
              <button
                class="sc-hKMtZM fntRFW"
              >
                <p
                  class="sc-dkzDqf gsGEgs"
                >
                  Videos
                </p>
              </button>
            </div>
            <div
              class="sc-iAvgwm erFLtt"
            >
              <div
                class="sc-jOrMOR hgyjza"
              >
                <div
                  class="sc-ciZhAO eOOvXq"
                >
                  <svg
                    class="sc-jdAMXn hGPkpt"
                    viewBox="25 25 50 50"
                  >
                    <circle
                      class="sc-bZnhIo fUqopE"
                      cx="50"
                      cy="50"
                      fill="none"
                      r="20"
                      stroke-miterlimit="10"
                      stroke-width="5"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div
              class="sc-iAvgwm BFhG"
            >
              <div
                class="sc-jOrMOR hgyjza"
              >
                <div
                  class="sc-ciZhAO eOOvXq"
                >
                  <svg
                    class="sc-jdAMXn hGPkpt"
                    viewBox="25 25 50 50"
                  >
                    <circle
                      class="sc-bZnhIo fUqopE"
                      cx="50"
                      cy="50"
                      fill="none"
                      r="20"
                      stroke-miterlimit="10"
                      stroke-width="5"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="sc-efBctP edPByx"
      >
        <div
          class="sc-kDDrLX uIMVe"
        >
          <div
            class="sc-iqcoie cVepWP"
          >
            <input
              class="sc-bczRLJ dldCsv"
              placeholder="search or paste a link"
              value=""
            />
          </div>
          <div
            class="sc-eCYdqJ FxPOW"
          >
            <button
              class="sc-hKMtZM fntRFW"
            >
              <svg
                class="sc-gsnTZi gSmFuM"
              >
                <use
                  href="http://localhost/collectSounds/icons.svg#filters"
                />
              </svg>
              <p
                class="sc-dkzDqf gsGEgs"
              >
                Filters
              </p>
            </button>
            <div
              class="sc-jSMfEi jmXqhj"
            >
              <div
                class="sc-iBkjds envIss"
              >
                <button
                  class="sc-gKXOVf feeGsc"
                >
                  <p
                    class="sc-dkzDqf jHUImv"
                  >
                    Addition
                  </p>
                </button>
                <ul
                  class="sc-ftvSup hiVoSg"
                >
                  <li>
                    <button
                      class="sc-gKXOVf feeGsc"
                    >
                      <p
                        class="sc-dkzDqf jHUImv"
                      >
                        Date
                      </p>
                    </button>
                  </li>
                  <li>
                    <button
                      class="sc-gKXOVf feeGsc"
                    >
                      <p
                        class="sc-dkzDqf jHUImv"
                      >
                        Title
                      </p>
                    </button>
                  </li>
                </ul>
              </div>
              <div
                class="sc-iBkjds envIss"
              >
                <button
                  class="sc-gKXOVf feeGsc"
                >
                  <p
                    class="sc-dkzDqf jHUImv"
                  >
                    All
                  </p>
                </button>
                <ul
                  class="sc-ftvSup hiVoSg"
                >
                  <li>
                    <button
                      class="sc-gKXOVf feeGsc"
                    >
                      <p
                        class="sc-dkzDqf jHUImv"
                      >
                        Video
                      </p>
                    </button>
                  </li>
                  <li>
                    <button
                      class="sc-gKXOVf feeGsc"
                    >
                      <p
                        class="sc-dkzDqf jHUImv"
                      >
                        Playlist
                      </p>
                    </button>
                  </li>
                  <li>
                    <button
                      class="sc-gKXOVf feeGsc"
                    >
                      <p
                        class="sc-dkzDqf jHUImv"
                      >
                        Channel
                      </p>
                    </button>
                  </li>
                </ul>
              </div>
              <div
                class="sc-iBkjds envIss"
              >
                <button
                  class="sc-gKXOVf feeGsc"
                >
                  <p
                    class="sc-dkzDqf jHUImv"
                  >
                    Any
                  </p>
                </button>
                <ul
                  class="sc-ftvSup hiVoSg"
                >
                  <li>
                    <button
                      class="sc-gKXOVf feeGsc"
                    >
                      <p
                        class="sc-dkzDqf jHUImv"
                      >
                        Long
                      </p>
                    </button>
                  </li>
                  <li>
                    <button
                      class="sc-gKXOVf feeGsc"
                    >
                      <p
                        class="sc-dkzDqf jHUImv"
                      >
                        Medium
                      </p>
                    </button>
                  </li>
                  <li>
                    <button
                      class="sc-gKXOVf feeGsc"
                    >
                      <p
                        class="sc-dkzDqf jHUImv"
                      >
                        Short
                      </p>
                    </button>
                  </li>
                </ul>
              </div>
              <div
                class="sc-jqUVSM hgFwMx"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-cTQhss fRlOMF"
      >
        <div>
          <div
            class="sc-iTONeN kTEwDW"
          >
            <button
              class="sc-hKMtZM ixzCuJ"
            >
              <p
                class="sc-dkzDqf gsGEgs"
              >
                All
              </p>
            </button>
            <button
              class="sc-hKMtZM fntRFW"
            >
              <p
                class="sc-dkzDqf gsGEgs"
              >
                Videos
              </p>
            </button>
          </div>
          <div
            class="sc-iAvgwm erFLtt"
          >
            <div
              class="sc-jOrMOR hgyjza"
            >
              <div
                class="sc-ciZhAO eOOvXq"
              >
                <svg
                  class="sc-jdAMXn hGPkpt"
                  viewBox="25 25 50 50"
                >
                  <circle
                    class="sc-bZnhIo fUqopE"
                    cx="50"
                    cy="50"
                    fill="none"
                    r="20"
                    stroke-miterlimit="10"
                    stroke-width="5"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="sc-iAvgwm BFhG"
          >
            <div
              class="sc-jOrMOR hgyjza"
            >
              <div
                class="sc-ciZhAO eOOvXq"
              >
                <svg
                  class="sc-jdAMXn hGPkpt"
                  viewBox="25 25 50 50"
                >
                  <circle
                    class="sc-bZnhIo fUqopE"
                    cx="50"
                    cy="50"
                    fill="none"
                    r="20"
                    stroke-miterlimit="10"
                    stroke-width="5"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
